### PR TITLE
Visual Studio: use the same warnings as in the CI

### LINKED ISSUE #125: Use the same language standard on every platform
In visual studio the language standatd is the `Legacy MSVC`, we should set it to c11 as we do with the [CI](https://github.com/cubzh/cubzh/blob/main/core/tests/cmake/CMakeLists.txt#L21)

image

### PR DESCRIPTION
**Pull request by XLegland**
Contributes to #125

### CODE REVIEW

File: core/tests/visual_studio/Cubzh Core Unit Tests.vcxproj [hunk 1]
```diff
@@ -243,13 +243,18 @@
   </ItemDefinitionGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
     <ClCompile>
-      <WarningLevel>Level4</WarningLevel>
+      <WarningLevel>EnableAllWarnings</WarningLevel>
       <FunctionLevelLinking>true</FunctionLevelLinking>
       <IntrinsicFunctions>true</IntrinsicFunctions>
       <SDLCheck>true</SDLCheck>
-      <PreprocessorDefinitions>NDEBUG;_USE_MATH_DEFINES;_CRT_SECURE_NO_WARNINGS;_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <PreprocessorDefinitions>DEBUG;NDEBUG;_USE_MATH_DEFINES;_CRT_SECURE_NO_WARNINGS;_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```
Comment (gdevillele) [inline core/tests/visual_studio/Cubzh Core Unit Tests.vcxproj:250]: have `DEBUG` and `NDEBUG` in the same configuration doesn't make sense.

**Review by gdevillele — COMMENTED** [review]
one comment :)

**Review by ArthurVoxowl — DISMISSED** [review]
Thanks!

File: core/tests/visual_studio/Cubzh Core Unit Tests.vcxproj [hunk 2]
```diff
@@ -243,13 +243,18 @@
   </ItemDefinitionGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
     <ClCompile>
-      <WarningLevel>Level4</WarningLevel>
+      <WarningLevel>EnableAllWarnings</WarningLevel>
       <FunctionLevelLinking>true</FunctionLevelLinking>
       <IntrinsicFunctions>true</IntrinsicFunctions>
       <SDLCheck>true</SDLCheck>
-      <PreprocessorDefinitions>NDEBUG;_USE_MATH_DEFINES;_CRT_SECURE_NO_WARNINGS;_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <PreprocessorDefinitions>DEBUG;_USE_MATH_DEFINES;_CRT_SECURE_NO_WARNINGS;_CONSOLE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```
Comment (gdevillele) [inline core/tests/visual_studio/Cubzh Core Unit Tests.vcxproj:250]: it's the `Release` config, so it should define `NDEBUG` and not `DEBUG`

**Review by gdevillele — COMMENTED** [review]
another fix needed please

**Review by gdevillele — DISMISSED** [review]
LGTM 👍

**Review by gdevillele — APPROVED** [review]

**Pull request merged**